ts/components/ui/ListItemAction.tsx

```diff
@@ -0,0 +1,222 @@
+import { isDesignSystemEnabledSelector } from "../../store/reducers/persistedPreferences";
+
+export type ListItemAction = WithTestID<{
+  label: string;
+  variant: "primary" | "danger";
+  onPress: (event: GestureResponderEvent) => void;
+  icon?: IOIcons;
+  // Accessibility
+  accessibilityLabel: string;
+}>;
+
+const styles = StyleSheet.create({
+  label: {
+    fontSize: 16,
+    lineHeight: 20,
+    ...makeFontStyleObject("Regular", false, "ReadexPro")
+  },
+  /* REMOVE_LEGACY_COMPONENT: Start ▶ */
+  labelLegacy: {
+    fontSize: 18,
+    lineHeight: 24,
+    ...makeFontStyleObject("SemiBold", false, "TitilliumWeb")
+  }
+  /* REMOVE_LEGACY_COMPONENT: End ▶ */
+});
+
+export const ListItemAction = ({
+  variant,
+  label,
+  onPress,
+  icon,
+  accessibilityLabel,
+  testID
+}: ListItemAction) => {
+  const isDesignSystemEnabled = useIOSelector(isDesignSystemEnabledSelector);
+  const isPressed: Animated.SharedValue<number> = useSharedValue(0);
+
+  const theme = useIOTheme();
+
+  const mapBackgroundStates: Record<string, string> = {
+    default: hexToRgba(IOColors[theme["listItem-pressed"]], 0),
+    pressed: IOColors[theme["listItem-pressed"]]
+  };
+
+  /* REMOVE_LEGACY_COMPONENT: Start ▶ */
+  const mapLegacyForegroundColor: Record<
+    NonNullable<ListItemAction["variant"]>,
+    IOColors
+  > = {
+    primary: "blue",
+    danger: "error-850"
+  };
+  /* REMOVE_LEGACY_COMPONENT: End ▶ */
+
+  const mapForegroundColor: Record<
+    NonNullable<ListItemAction["variant"]>,
+    string
+  > = {
+    primary: IOColors[theme["interactiveElem-default"]],
+    danger: IOColors[theme.errorText]
+  };
+
+  // Scaling transformation applied when the button is pressed
+  const animationScaleValue = IOScaleValues?.basicButton?.pressedState;
+
+  const progressPressed = useDerivedValue(() =>
+    withSpring(isPressed.value, IOSpringValues.button)
+  );
+
+  // Interpolate animation values from `isPressed` values
+  const animatedScaleStyle = useAnimatedStyle(() => {
+    const scale = interpolate(
+      progressPressed.value,
+      [0, 1],
+      [1, animationScaleValue],
+      Extrapolate.CLAMP
+    );
+
+    return {
+      transform: [{ scale }]
+    };
+  });
+
+  const animatedBackgroundStyle = useAnimatedStyle(() => {
+    const backgroundColor = interpolateColor(
+      progressPressed.value,
+      [0, 1],
+      [mapBackgroundStates.default, mapBackgroundStates.pressed]
+    );
+
+    return {
+      backgroundColor
+    };
+  });
+
+  const onPressIn = useCallback(() => {
+    // eslint-disable-next-line functional/immutable-data
+    isPressed.value = 1;
+  }, [isPressed]);
+  const onPressOut = useCallback(() => {
+    // eslint-disable-next-line functional/immutable-data
+    isPressed.value = 0;
+  }, [isPressed]);
+
+  /* ◀ REMOVE_LEGACY_COMPONENT: Start */
+  const LegacyListItemAction = () => (
+    <Pressable
+      onPress={onPress}
+      onPressIn={onPressIn}
+      onPressOut={onPressOut}
```

Comment:
In order to avoid what happened with the PR #4644, what do you think to trigger the `onPressOut` handler even for the event `onTouchEnd`?

```suggestion
      onPressOut={onPressOut}
      onTouchEnd={onPressOut}
```